data/types.go

```diff
@@ -147,3 +160,16 @@ func DefaultExpires(role string) time.Time {
 	}
 	return t.UTC().Round(time.Second)
 }
+
+type unmarshalledSignature Signature
+
+func (s *Signature) UnmarshalJSON(data []byte) error {
+	uSignature := unmarshalledSignature{}
+	err := json.Unmarshal(data, &uSignature)
+	if err != nil {
+		return err
+	}
+	uSignature.Method = SigAlgorithm(strings.ToLower(string(uSignature.Method)))
```

Comment:
I'll just point out this is exactly the problem I was alluding to that isn't solved by types. When deserializing we have no idea if the value is one for which we have the appropriate signature verifier. Without doing the types we wouldn't need the unmarshaller at all as this seems to be its only purpose (noting that we still need to do ToLower in the signing and verification methods because a user may have passing in a value from their own code, rather than it coming in through JSON).